### PR TITLE
Feature/complex schema field comparator modifications

### PR DESCRIPTION
**Pull request by Alesiks**

### CODE REVIEW

File: src/main/java/com/epam/eco/commons/avro/AvroUtils.java [hunk 1]
```diff
@@ -136,7 +138,7 @@ public static GenericRecord decodeRecordFromBinary(
                 reader = new GenericDatumReader<>(schema);
             }
 
-            return (GenericRecord)reader.read(null, decoder);
+            return (GenericRecord) reader.read(null, decoder);
```
Comment (andrei-tytsik) [inline src/main/java/com/epam/eco/commons/avro/AvroUtils.java]: please no such changes

File: src/main/java/com/epam/eco/commons/avro/AvroUtils.java [hunk 2]
```diff
@@ -289,7 +291,7 @@ public static Type typeOfGenericSchema(Object genericSchema) {
         Validate.notNull(genericSchema, "Generic schema is null");
 
         if (genericSchema instanceof Map) {
-            return Type.valueOf(((Map<String, Object>)genericSchema).
+            return Type.valueOf(((Map<String, Object>) genericSchema).
```
Comment (andrei-tytsik) [inline src/main/java/com/epam/eco/commons/avro/AvroUtils.java]: please no such changes

**Comment by andrei-tytsik** [comment]
Please rewrite using effectiveTypeXXX methods + modify test to cover all the edge cases
